Source/WebCore/editing/TextIterator.cpp

```diff
@@ -498,29 +498,29 @@ void TextIterator::advance()
 
         // find a new current node to handle in depth-first manner,
         // calling exitNode() as we come back thru a parent node
-        RefPtr next = m_handledChildren ? nullptr : firstChild(m_behaviors, *m_currentNode);
+        RefPtr next = m_handledChildren ? nullptr : firstChild(m_behaviors, *protectedCurrentNode());
         m_offset = 0;
         if (!next) {
-            next = nextSibling(m_behaviors, *m_currentNode);
+            next = nextSibling(m_behaviors, *protectedCurrentNode());
             if (!next) {
-                bool pastEnd = nextNode(m_behaviors, *m_currentNode) == m_pastEndNode;
-                RefPtr parentNode = parentNodeOrShadowHost(m_behaviors, *m_currentNode);
+                bool pastEnd = nextNode(m_behaviors, *protectedCurrentNode()) == m_pastEndNode;
+                RefPtr parentNode = parentNodeOrShadowHost(m_behaviors, *protectedCurrentNode());
                 while (!next && parentNode) {
                     if ((pastEnd && parentNode == m_endContainer.get()) || isDescendantOf(m_behaviors, *m_endContainer, *parentNode))
                         return;
                     bool haveRenderer = isRendererVisible(m_currentNode->renderer(), m_behaviors);
-                    RefPtr exitedNode = m_currentNode.get();
+                    auto exitedNode = protectedCurrentNode();
                     m_currentNode = parentNode;
```

Comment:
Perhaps capturing the result of `protectedCurrentNode()` after this assignment would allow you to resume the same object in the rest of this method without having to create new Ref's at each call site?